packages/dds/tree/src/test/domains/json/citm.ts

```diff
@@ -0,0 +1,663 @@
+    seatCategoryNames: Record<string, string>,
+    audienceSubCategoryNames: Record<string, string>,
+    areaNames: Record<string, string>,
+) {
+    // Semi-Randomly select topic Id's
+    const eventTopicIdSet = new Set<number>();
+    const numTopicsToInclude = random.integer(0, Math.min(4, topicIds.length));
+    for (let j = 0; j < numTopicsToInclude; j++) {
+        let topicIdIndex = random.integer(0, topicIds.length - 1);
+        let topicIdToAdd = Number.parseInt(topicIds[topicIdIndex], 10);
+        // If random selection picks a previously used topicId
+        // then we increment forwards once until an unencountered id appears
+        while (eventTopicIdSet.has(topicIdToAdd)) {
+            topicIdIndex += 1;
+            if (topicIdIndex > topicIds.length - 1) {
+                topicIdIndex = topicIdIndex % topicIds.length;
+            }
+            topicIdToAdd = Number.parseInt(topicIds[topicIdIndex], 10);
+        }
+        eventTopicIdSet.add(topicIdToAdd);
+    }
+
+    // Semi-Randomly select subTopic Id's under each event topic
+    const eventSubTopicIdSet = new Set<number>();
+    if (numTopicsToInclude > 0) {
+        let processedTopicIdCount = 0;
+        eventTopicIdSet.forEach((topicId) => {
+            const topicSubTopicIds = topicSubTopics[`${topicId}`];
+            // This reserves atleast 1 subtopicId to be added for each topic id.
+            const unprocessedTopicIds = eventTopicIdSet.size - processedTopicIdCount;
+            const numSubTopicsToInclude = random.integer(
+                1,
+                Math.min(topicSubTopicIds.length, unprocessedTopicIds),
+            );
+            for (let x = 0; x < numSubTopicsToInclude; x++) {
+                let subTopicIndex = random.integer(0, topicSubTopicIds.length - 1);
+                let subTopicIdToAdd = topicSubTopicIds[subTopicIndex];
+                // If random selection picks a previously used topicId
+                // then we increment forwards once until an unencountered id appears
+                while (eventSubTopicIdSet.has(subTopicIdToAdd)) {
+                    subTopicIndex += 1;
+                    if (subTopicIndex > topicSubTopicIds.length - 1) {
+                        subTopicIndex = subTopicIndex % topicSubTopicIds.length;
+                    }
+                    subTopicIdToAdd = topicSubTopicIds[subTopicIndex];
+                }
+                eventSubTopicIdSet.add(subTopicIdToAdd);
+            }
+            processedTopicIdCount++;
+        });
+    }
+
+    // All logo strings in the original follow this pattern. and have a 48.913% chance of being null
+    const logo = random.bool(0.48913)
+        ? `/images/UE0AAAAACEK${getRandomEnglishString(random, true, 2, 2)}QAAAAVDSVRN`
+        : null;
+
+    const event: CitmCatalog.Event = {
+        description: null,
+        id: idNumberCounter,
+        logo,
+        name: ORIGINAL_EVENT_NAMES[random.integer(0, ORIGINAL_EVENT_NAMES.length - 1)],
```

Comment:
Skimming the original, I suspect that event names is the "thing to scale", while categories, subtopics, seating area, etc. can be chosen from a fixed set of enum-like values.

Reply:
Could you elaborate on why the keyspace for event names should be scaled? (Json size, etc..) There are effectively just descriptions on each event.

Reply:
Let's take it as is and I can tweak it after the fact.